### PR TITLE
userservice unit test.

### PR DESCRIPTION
**Pull request by dpoarchmicrosoft**

### CODE REVIEW

**Review by namalu — CHANGES_REQUESTED** [review]

File: src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            _googleFitClient = Substitute.For<IGoogleFitClient>();
+            _authService = Substitute.For<IGoogleFitAuthService>();
+            _usersTableRepository = Substitute.For<IUsersTableRepository>();
+            _logger = Substitute.For<ILogger<UsersService>>();
+
+            _usersService = new UsersService(_usersTableRepository, _googleFitClient, _usersKeyvaultRepository, _authService, _logger);
+        }
+
+        [Fact]
+        public void GivenRequestHandledAndExceptionIsThrown_WhenAccessCodeIsNotValid_ReturnsExceptionTokenResponseEmpty()
```
[the remark this answers is not in the document]
Comment (dpoarchmicrosoft) [reply src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs]: update code

File: src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            Task<Exception> exception = Assert.ThrowsAsync<Exception>(() => _usersService.Initiate(Arg.Any<string>(), Arg.Any<CancellationToken>()));
```
[the remark this answers is not in the document]
Comment (dpoarchmicrosoft) [reply src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs]: update code

File: src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            Assert.Equal(exception.Result.Message, new Exception("Token response empty").Message);
+        }
+
+        [Fact]
+        public void GivenRequestHandledAndExceptionIsThrown_WhenEmailResponseIsEmpty_ReturnsExceptionTokenResponseEmpty()
+        {
+            _authService.AuthTokensRequest(Arg.Any<string>(), Arg.Any<CancellationToken>()).Returns(new AuthTokensResponse());
+            _googleFitClient.MyEmailRequest(Arg.Any<string>(), Arg.Any<CancellationToken>()).Throws(new Exception("Email response empty"));
```
[the remark this answers is not in the document]
Comment (dpoarchmicrosoft) [reply src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs]: update code

File: src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+using GoogleFitOnFhir.Repositories;
+using GoogleFitOnFhir.Services;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using Xunit;
+
+namespace GoogleFitOnFhir.UnitTests
+{
+    public class GoogleFitUserServiceTests
```
[the remark this answers is not in the document]
Comment (dpoarchmicrosoft) [reply src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs]: update code

File: src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        private readonly IUsersKeyVaultRepository _usersKeyvaultRepository;
+        private readonly IGoogleFitAuthService _authService;
+
+        public GoogleFitUserServiceTests()
+        {
+            _usersKeyvaultRepository = Substitute.For<IUsersKeyVaultRepository>();
+            _googleFitClient = Substitute.For<IGoogleFitClient>();
+            _authService = Substitute.For<IGoogleFitAuthService>();
+            _usersTableRepository = Substitute.For<IUsersTableRepository>();
+            _logger = Substitute.For<ILogger<UsersService>>();
```
[the remark this answers is not in the document]
Comment (dpoarchmicrosoft) [reply src/GoogleFitOnFhir.UnitTests/GoogleFitUserServiceTests.cs]: update code

**Comment by namalu** [comment]
Obsolete

**Pull request closed without merging**